src/go/util/url_util_test.go

```diff
@@ -446,14 +446,37 @@ func TestWildcardMatcherForPath(t *testing.T) {
 			wantMatcher: "",
 		},
 		{
-			desc:        "Multiple path params",
+			desc:        "Path params with fieldpath-only bindings",
 			uri:         "/shelves/{shelf_id}/books/{book.id}",
-			wantMatcher: `/shelves/[^\/]+/books/[^\/]+$`,
+			wantMatcher: `^/shelves/[^\/]+/books/[^\/]+$`,
 		},
 		{
-			desc:        "Path param with wildcard",
+			desc:        "Path param with wildcard segments",
+			uri:         "/test/*/test/**",
```

Comment:
Do we have url pattern conflict check for duplicate regex?
/test/*
/test/{x=2}

I remember api compiler has this check but it is good to have on control plane if we don't have.

Reply:
Path matcher filter will reject the config if there are duplicated http templated paths. We don't need another validation for the hacky regex conversion in config generator, since the end result to the user will be the same.

Reply:
To clarify, duplicated path templates will account for the two cases you provided above.